### ISSUE TITLE
Record configuration changes

### ISSUE THREAD

**Issue by garethbowen** [medic/medic-webapp#1616]
As our app becomes more and more configurable (xforms, nools, etc), the configuration becomes more and more critical. We need to record when changes are made to configuration (app_settings, forms, etc) and make it easy to roll back if necessary. Ideally we'd also capture who made the change.

**Comment by alxndrsn** [comment]
Does couchdb already store all revisions of the ddoc?

**Comment by garethbowen** [comment]
Yes but these could be compacted any time so we can't rely on it.

**Comment by dianabarsan** [comment]
I believe this is partially done through cht-conf now. And regardless of whether or not all angles are covered, this work should still be part of cht-conf, not cht-core. Closing.
